src/tools/rbd_mirror/LeaderWatcher.h

```diff
@@ -0,0 +1,63 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_RBD_MIRROR_LEADER_WATCHER_H
+#define CEPH_RBD_MIRROR_LEADER_WATCHER_H
+
+#include "librbd/Watcher.h"
+#include "tools/rbd_mirror/leader_watcher/Types.h"
+
+namespace rbd {
+namespace mirror {
+
+class LeaderWatcher : protected librbd::Watcher {
+public:
+  LeaderWatcher(librados::IoCtx &io_ctx, ContextWQ *work_queue);
+
+  int init();
+  void shut_down();
+
+  void notify_heartbeat(Context *on_finish);
+  void notify_lock_acquired(Context *on_finish);
+  void notify_lock_released(Context *on_finish);
+
+  virtual void handle_heartbeat(Context *on_ack) = 0;
```

Comment:
Nit: these methods are unused in this commit and refactored into protected, non-virtual methods in another commit